src/node_modules/logger.js

```diff
@@ -308,94 +351,81 @@ export default class Logger extends Writable {
 		return name ? [num, name] : [];
 	}
 
-	async [$getConnection](_resolve, _reject) {
+	async [$getConnection]() {
 		// The $connection property is a promise. On error, manual close, or
-		// timeout, it deletes itself.
-
-		if (this[$connection]) return this[$connection];
+		// inactivityTimeout, it deletes itself.
 
-		if (this[$failedAttempts]) {
-			await new Promise(resolve => {
-				setTimeout(resolve, Math.min(defaults.timeouts.connection.maxRetry, (this[$failedAttempts] ** 1.5) * 1000));
-			});
+		if (this[$connection]) {
+			return this[$connection];
+		} else {
+			DEBUG.log("No connection exists. Creating a new one.");
+			return this.getConnectionPromise();
 		}
+	}
 
-		this[$connection] = new Promise((resolve, reject) => {
-			resolve = _resolve || resolve;
-			reject = _reject || reject;
+	getConnectionPromise() {
+		// clear the state of previous reconnection and create a new one with a new connection promise.
+		if (this[$reconnection]) {
+			// destroy previous reconnect instance if it exists.
+			this[$reconnection].disconnect();
+			delete this[$reconnection];
+		}
 
-			if (this[$cannotConnect]) {
-				reject(new LogentriesError(text.cannotConnect()));
-				return;
-			}
+		this[$reconnection] = this[$reconnect]({
+			// all options are optional
+			initialDelay: this.reconnectInitialDelay,
+			maxDelay: this.reconnectMaxDelay,
+			strategy: this.reconnectBackoffStrategy,
+			failAfter: Infinity,
+			randomisationFactor: 0,
+			immediate: false
+		});
 
-			const opts = {
+		this[$connection] = new Promise((resolve, reject) => {
+			const connOpts = {
 				host: this.host,
 				port: this.port
 			};
 
-			let connection;
-			let resolved = false;
-
-			if (this.secure) {
-				connection = tls.connect(opts, () => {
-					if (!connection.authorized) {
-						this[$failedAttempts]++;
-						const errMsg = connection.authorizationError;
-						this.emit(new LogentriesError(text.authError(errMsg)));
-					} else if (tls && tls.CleartextStream &&
-						connection instanceof tls.CleartextStream)
-					// Your node is ancient!
-						this.emit('connect');
+			this[$reconnection].on('connect', function (con) {
+				DEBUG.log("Connected");
+				this.emit('connected');
+				con.on('timeout', () => {
+					//we listen to timeout event and close connection if it happens. see below.
+					this.emit("connection timeout");
```

Comment:
This event is emitted to $reconnection and used internally but I know what you mean. On it.